match/lib/match/git_helper.rb

```diff
@@ -156,5 +156,28 @@ def self.add_user_config(user_name, user_email)
         end
       end
     end
+
+    # Checks push permission to git repo
+    def self.repo_pushable?(path, branch = "master")
+      Dir.chdir(path) do
+        command = "GIT_TERMINAL_PROMPT=0 git push origin #{branch.shellescape} --dry-run"
+        FastlaneCore::CommandExecutor.execute(command: command,
+                                              print_all: FastlaneCore::Globals.verbose?,
+                                              print_command: FastlaneCore::Globals.verbose?)
+      end
+      return true
+    rescue => ex
+      UI.error("No permission to push...")
+      UI.error(ex)
```

Comment:
@KrauseFx 
`FastlaneCore::CommandExecutor` outputs
```
ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[18:06:01]: Exit status: 128
```

`repo_pushable?` outputs
```
[18:06:01]: No permission to push...
[18:06:01]: Exit status: 128
```

`check_push_repo_permission` outputs
```
[18:06:01]: You do not have push permission to git repository provided
[18:06:01]: _match_ needs to create a new certificate or provisioning profile, however without push access to the git repo, the generated certificate can't be stored properly, resulting in an unused certificate

[!] Please grant push access for the current git user to the git repo, so that _match_ can update and create certificates for you
```

Don't need duplicate `Exit status` errors?